src/server/ServerHandler.java

```diff
@@ -308,6 +308,48 @@ private PairingResponse handleUpdatePairing() {
         }
     }
 
+    /**
+     * Handles the acceptance of a pending game invitation by the opponent.
+     * This method retrieves the event associated with the given {@code eventId}
+     *
+     * @param eventId the unique identifier of the event (invitation) to be accepted
+     * @return a {@link Response} object indicating whether the acceptance was successful or failed,
+     * with an appropriate message
+     */
+
+    public Response handleAcceptInvitation(int eventId) throws SQLException {
+        // Retrieve the event from the database
+        Event event = DatabaseHelper.getInstance().getEvent(currentEventId);
+
+        // Check if the event exists
+        if (event == null) {
+            return new Response(ResponseStatus.FAILURE, "Event not found.");
+        }
+
+        // Check if the event is still pending
+        if (!event.getStatus().equals("PENDING")) {
+            return new Response(ResponseStatus.FAILURE, "This invitation is no longer available.");
+        }
+
+        // Check if the opponent is the current user
+        if (!event.getOpponent().equals(currentUsername)) {
+            return new Response(ResponseStatus.FAILURE, "You are not authorized to accept this invitation.");
+        }
+
+        // Abort any other pending invitations for this user
+        DatabaseHelper.abortAllUserEvents(currentUsername);
+
+        // Update event status to ACCEPTED
+        event.setStatus(Event.EventStatus.valueOf("ACCEPTED"));
```

Comment:
Use the enum value directly
```suggestion
        event.setStatus(Event.EventStatus.ACCEPTED);
```